### ISSUE TITLE
Remove SMART authorization module redirectUri default when the redirectUri is not present

### ISSUE THREAD

**Issue by mseckykoebel** [flexpa/sero#92]
When building with the SMART auth module, there are instances where the default `redurectUri`, which is set to `http://localhost:8080${redirectPath}`, becomes the `recurectUri` when it shouldn't. Better behavior for this would be to remove this default and throw an error if the `redirectUri` is not provided properly when the module is initialized in a new project.

In short, the default behavior of the module should be to throw an error if the `redirectUri` is not provided instead of providing a default `redirectUri`.

**Comment by jdjkelly** [comment]
```typescript
  const { name, credentials, scope, redirectHost } = options

  const prefix = options.prefix = "/smart";
  const authorizeParams = options.authorizeParams || {}
  const authorizePath = options.authorizePath || `${prefix}/${name.toLowerCase()}/auth`
  const redirectPath = options.redirectPath || `${prefix}/${name.toLowerCase()}/redirect`
  const redirectUri = `${redirectHost}${redirectPath}`
````

Add a new mandatory field called redirectHost to the options and this should work better.

**Comment by jdjkelly** [comment]
```
export interface SmartAuthProvider {
  name: string;
  scope: string[]; // @todo this could be typed to the FHIR spec
  credentials: SmartAuthCredentials;
  redirectHost: string;
  redirectPath?: string; 
  authorizePath?: string;
  authorizeParams?: Object; // @todo
  prefix?: string;
}
```

**Comment by jdjkelly** [comment]
From the spec: http://www.hl7.org/fhir/smart-app-launch/

> No matter how an app registers with an EHR’s authorization service, at registration time every SMART app must:

> * Register zero or more fixed, fully-specified launch URL with the EHR’s authorization server
> * Register one or more fixed, **fully-specified redirect_uris** with the EHR’s authorization server. Note: In the case of native clients following the OAuth 2.0 for Native Apps specification (RFC 8252), it may be appropriate to leave the port as a dynamic variable in an otherwise fixed redirect URI.

So kind of makes sense to enforce this at the config level too

**Comment by mseckykoebel** [comment]
Closed with #94.
